src/EnergyPlus/UnitarySystem.cc

```diff
@@ -2883,4288 +2899,4153 @@ namespace UnitarySystems {
         //    UnitarySystemNumericFields.deallocate(); // remove temporary array for field names at end of sizing
     }
 
-    void UnitarySys::getUnitarySystemInputData(
-        EnergyPlusData &state, std::string const &objectName, bool const ZoneEquipment, int const ZoneOAUnitNum, bool &errorsFound)
+    void UnitarySys::processInputSpec(EnergyPlusData &state,
+                                      const UnitarySysInputSpec &input_data,
+                                      int sysNum,
+                                      bool &errorsFound,
+                                      bool const ZoneEquipment,
+                                      int const ZoneOAUnitNum)
     {
 
-        static std::string const getUnitarySystemInput("getUnitarySystemInputData");
         static std::string const unitarySysHeatPumpPerformanceObjectType("UnitarySystemPerformance:Multispeed");
 
-        std::string cCurrentModuleObject = "AirLoopHVAC:UnitarySystem";
-
-        auto const instances = state.dataInputProcessing->inputProcessor->epJSON.find(cCurrentModuleObject);
-        if (instances == state.dataInputProcessing->inputProcessor->epJSON.end()) {
-            ShowSevereError(state, "getUnitarySystemInputData: did not find AirLoopHVAC:UnitarySystem object in input file. Check inputs");
-            errorsFound = true;
-        } else {
-            auto &instancesValue = instances.value();
-            for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
+        std::string cCurrentModuleObject("AirLoopHVAC:UnitarySystem");
+        std::string thisObjectName = input_data.name;
+
+        std::string loc_AirInNodeName = input_data.air_inlet_node_name;
+        if (state.dataUnitarySystems->getInputOnceFlag)
```

Comment:
It would be really good to go ahead and put braces around this call to GetOnlySingleNode even though it's not syntactically necessary.  Extra indentation can cause weirdness.  Nothing to necessarily change right now though.